### PR TITLE
Authenticated check with traefik instead of auth-poxy.

### PR DESCRIPTION
**Pull request by oruharo**
- Authenticated check with traefik instead of auth-poxy.
- Share user authentication for workspace with dashboard.
- Bump go packages.

TODO:
- Document Changes.
- Authentication manifest is generated automatically instead of writing it in the tempalate.

### CODE REVIEW

**Review by jlandowner — APPROVED** [review]

File: traefik/plugins/cosmo-workspace/cosmoauth/cosmoauth_middleware.go [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+		p.redirectToLoginPage(w, r)
+		return
+	}
+
+	sesInfo := session.Get(ses)
+	LoggerDEBUG.Print("get session.", " userName=", sesInfo.UserName, " deadline=", sesInfo.Deadline)
+
+	// check user name is owner's
+	userName := r.Header.Get("X-Cosmo-UserName")
+	if userName != "" && sesInfo.UserName != userName {
```
Comment (jlandowner) [inline traefik/plugins/cosmo-workspace/cosmoauth/cosmoauth_middleware.go:105]:
```suggestion
	// TODO: improvement
	if userName != "" && sesInfo.UserName != userName {
```

File: internal/dashboard/root.go [hunk 2]
```diff
@@ -0,0 +1,227 @@
...
+		return fmt.Errorf("validation error: %w", err)
+	}
+	if err := o.Complete(cmd, args); err != nil {
+		return fmt.Errorf("invalid options: %w", err)
+	}
+	return nil
+}
+
+func (o *options) Validate(cmd *cobra.Command, args []string) error {
+	if o.CookieDomain == "" {
```
Comment (jlandowner) [inline internal/dashboard/root.go:116]:
```suggestion
	// TODO: validate length
	if o.CookieDomain == "" {
```

File: internal/dashboard/root.go [hunk 3]
```diff
@@ -0,0 +1,227 @@
...
+	klog.InitFlags(goflags)
+	o.ZapOpts.BindFlags(goflags)
+	ctrl.RegisterFlags(goflags)
+	rootCmd.Flags().AddGoFlagSet(goflags)
+
+	rootCmd.PersistentFlags().Int64Var(&o.ResponseTimeoutSeconds, "timeout-seconds", 3, "Timeout seconds for response")
+	rootCmd.PersistentFlags().Int64Var(&o.GracefulShutdownSeconds, "graceful-shutdown-seconds", 10, "Graceful shutdown seconds")
+	rootCmd.PersistentFlags().StringVar(&o.StaticFileDir, "serve-dir", "/app/public", "Static file dir to serve")
+	rootCmd.PersistentFlags().StringVar(&o.CookieDomain, "cookie-domain", "", "Cookie domain name")
+	rootCmd.PersistentFlags().StringVar(&o.CookieHashKey, "cookie-hashkey", "", "Cookie hashkey")
```
Comment (jlandowner) [inline internal/dashboard/root.go:93]: TODO: better that sensitive values are given by secret env

File: internal/dashboard/root.go (hunk 2 above)
Comment (oruharo) [reply internal/dashboard/root.go:117]: If cookie-domain is not specified, cookies can only be enabled between the same hosts.
This is assuming that the host is localhost, etc.

File: internal/dashboard/root.go (hunk 3 above)
Comment (oruharo) [reply internal/dashboard/root.go:93]: Give these values in secret env.
Is there another way?
https://github.com/cosmo-workspace/cosmo/blob/main/charts/cosmo-traefik/templates/cosmo-auth-middleware.yaml

**Pull request merged**